### DISCUSSION TITLE
Why are some components that doesnt has the type 'variant' in their props?

### DISCUSSION THREAD

**Discussion by TigreDev** [GeekyAnts/NativeBase#5051] [Q&A]
I am starting a new project with NativeBase, I have to say that it is incredibly comfortable to use, really intuitive and very easy to adapt.

I found that components can use variants, extending the component in the theme to be able to reuse some component styles without the need to generate a new atom component.

So I test this to make a layout stack called mainContainer:

```
  const theme = extendTheme({
    components: {
      Stack: {
        variants: {
          mainContainer: {
            direction: 'column',
            space: 6,
            px: 6,
            flex: 1,
            pt: 10,
            bgColor: '#fff',
            _dark: {
              bgColor: '#000',
            },
          },
        },
      },
    },
  });
```

But then I found that there are components that, although they support the use of variants (since this style works) have not included in their prop types the variant, so you get a ts error when the variant prop is included.

```
<Stack variant="mainContainer">
       ...
</Stack>
```

![image](https://user-images.githubusercontent.com/67019763/171265254-175c16f5-5125-4299-bbd9-f9ce4c4997fd.png)

I'm doing something wrong?
It's intended to be a ts error on this components?

**Comment by TigreDev** [comment]
I guess this is intended to be this way?

**Reply by sanna231** [reply]
True, but if you have extended your theme ideally it should add variant type in the types of that component, we will definitely consider this for future releases.
